starter_code/src/movies.js

```diff
@@ -1,15 +1,148 @@
+function orderByYear(movies) {
+   
+    let newMovies = [...movies];
+    if ( movies.length == 0  ){
+       return newMovies ;
+    }
+    
+    newMovies.sort(function(moviea, movieb) { 
+        return moviea.year - movieb.year  ;
+    });
+    
+    newMovies.sort((a, b) => (a.year > b.year) ? 1 : (a.year === b.year) ? ((a.title > b.title) ? 1 : -1) : -1 )
+     
+    //console.log(newMovies); 
+    return newMovies;    
+}    
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+    function howManyMovies(movies) {
 
+        if (movies.length == 0){
+            return 0;
+        }      
+
+        const moviesNumbers = movies.filter(movie => movie.director == "Steven Spielberg" );
+
+        const moviesNumbers2 = moviesNumbers.filter(movie2 => movie2.genre.indexOf('Drama') > -1 );
+
+        return moviesNumbers2.length; 
+
+    } 
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+    function orderAlphabetically(movies) {
+        
+        let newMovies = [...movies];
+        newMovies.sort((a,b) =>{
+            if (a.title.toLowerCase() > b.title.toLowerCase() ) return 1;   
+            if (a.title.toLowerCase() < b.title.toLowerCase() ) return -1;
+            return 0;
+        })
+        
+        let moviesTitle = [];
+        newMovies.forEach(element => moviesTitle.push(element.title));
+        
+        return moviesTitle.slice(0, 20) ;
+    };
 
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
+    function ratesAverage(movies) {
+        
+        if (movies.length == 0){
+            return 0;
+        }      
+        
+        let movieNumber = 0;        
+        var total = movies.reduce(function(accumulator, currentValue){            
+            if ( !isNaN(currentValue.rate) ){
+                movieNumber = movieNumber+1; 
+                return accumulator + currentValue.rate;
+            } else {
+                return accumulator ;
+            }
+        }, 0);
+        //console.log(movieNumber, total)
+        return Math.round((total / movies.length)*100)/100 ;
+    }
 
 // Iteration 5: Drama movies - Get the average of Drama Movies
+    function dramaMoviesRate(movies){
+
+        //exclure drama//
+        const moviesDrama = movies.filter(movie => movie.genre == "Drama" );
+        let total = ratesAverage(moviesDrama);
+        return total ;
+    } 
 
 // Iteration 6: Time Format - Turn duration of the movies from hours to minutes
+    function turnHoursToMinutes(movies) {
+        
+        const movies1 = movies.map(movie => {
+            return {
+              name: movie.name,
+              title:  movie.title,
+              year: movie.year,
+              director: movie.director,
+              duration: Number(convertDuration(movie.duration)),
+              genre:movie.genre,
+              rate: movie.rate,
+            }
+          });
```

Comment:
Nice way of creating a new object, to make your life easier you could have done it this way:

Here is a simpler solution since you want to get all the keys and values but only want to change one key / value pair 

```
 function turnHoursToMinutes(movies) {
        
        const movies1 = movies.map(movie => {
            return {
                ...movie,
              duration: Number(convertDuration(movie.duration)),
           
            }
          });
          
        return movies1 ;
    }
```